packages/visx-responsive/src/resizeObserver.ts

```diff
@@ -0,0 +1,36 @@
+// @TODO remove all these types when upgraded to TS 4 which has its own declaration
+interface ResizeObserverEntry {
+  contentRect: {
+    left: number;
+    top: number;
+    width: number;
+    height: number;
+  };
+}
+
+type ResizeObserverCallback = (entries: ResizeObserverEntry[], observer: ResizeObserver) => void;
+
+export declare class ResizeObserver {
+  constructor(callback: ResizeObserverCallback);
+  observe(target: Element, options?: any): void;
+  unobserve(target: Element): void;
+  disconnect(): void;
+  static toString(): string;
+}
+
+export interface ResizeObserverPolyfill {
+  new (callback: ResizeObserverCallback): ResizeObserver;
+}
+
+interface PrivateWindow {
+  ResizeObserver: ResizeObserverPolyfill;
+}
+
+let resizeObserverPolyfill: ResizeObserverPolyfill | undefined;
+
+export const getResizeObserver = () =>
+  resizeObserverPolyfill || (window as unknown as PrivateWindow).ResizeObserver;
+
+export const setResizeObserverPolyfill = (polyfill: ResizeObserverPolyfill | undefined) => {
```

Comment:
I have mixed feelings about this approach, I wouldn't want the singleton to cause issues e.g., with SSR/client mismatches or state leaking. Since it's opt-in it could still be okay, but maybe we shouldn't mark the existing config/props as deprecated? Sorry to make you undo that.

Reply:
Yeah I think I will just revert it. The proposed solution is probably quite bad in general, a React context would be a better option but I'm wondering if it's worth the effort, polyfilling ResizeObserver feels like something that might be useful only in tests considering current browser support of this feature, but even then majority of users would probably be happy with a simple mock attached to window with empty methods.

Reply:
yeah agree, hopefully this is temporary and we can remove it entirely soon 👍